avro-schema-common/src/main/java/io/micronaut/avro/model/AvroSchema.java

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.avro.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.serde.annotation.Serdeable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Represents an Avro schema, which defines the structure of data in Avro format.
+ *
+ * @author Ali Linaboui
+ * @since 1.0
+ */
+@Internal
+@Serdeable
+public final class AvroSchema {
```

Comment:
consider adding the sourcegen module and using `@Builder` and `@Wither` with records